### PR TITLE
timer: use read/write lock for timer's memory store

### LINKED ISSUE #44720: Use RW Lock for timer's memory store

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #44720

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by bb7133 — APPROVED** [review]
LGTM

File: timer/api/mem_store.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func (s *memoryStoreCore) Close() {
 }
 
 type memTimerWatchEventNotifier struct {
-	mu       sync.Mutex
+	mu       sync.RWMutex
```
Comment (hawkingrei) [inline timer/api/mem_store.go:180]: Not using a write lock?

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**